src/pages/MainPage/index.jsx

```diff
@@ -1,14 +1,67 @@
-import { QuoteCard } from '../../Components/QuoteCard'; 
-import { Button } from '../../Components/Button';
+import { QuoteCard } from "../../components/QuoteCard";
+import { Button } from "../../components/Button";
+import { useQuotesContext } from "../../QuotesContextProvider";
+import {
+  useQuoteIndexContext,
+  useQuoteIndexDispatchContext,
+} from "../../QuoteIndexContextProvider";
+import { useState } from "react";
+import { useFavoritesContext } from "../../FavoritesContext";
 
-export const MainPage = ({quote , author ,likeCount, handleNextQuoteClick , handleLikeQuoteClick}) =>{
-  return(
+export const MainPage = () => {
+  const quotes = useQuotesContext();
+  const currentIndex = useQuoteIndexContext();
+  const dispatchQuoteIndex = useQuoteIndexDispatchContext();
+  const [updatedQuotes, setUpdatedQuotes] = useState(quotes);
+  const { favorites, handleAddToFavorites } = useFavoritesContext();
+
+  function handleNextQuoteClick() {
+    const randomIndex = Math.floor(Math.random() * quotes.length);
+    dispatchQuoteIndex(randomIndex);
+  }
+  function handleLike() {
+    const newQuotes = updatedQuotes.map((quote, index) => {
+      if (currentIndex === index) {
+        return { ...quote, likeCount: quote.likeCount + 1 };
+      } else return quote;
+    });
+    setUpdatedQuotes(newQuotes);
```

Comment:
Use setQuotes from `useQuotesDispatchContext` here instead of `setUpdatedQuotes` from the local state.